src/osiris_log.erl

```diff
@@ -3241,9 +3347,104 @@ list_dir(Dir) ->
             [list_to_binary(F) || F <- Files]
     end.
 
+ra_read(ReadPos, Len, #ra{buf = {Pos, Data}})
+  when ReadPos >= Pos andalso
+       ReadPos + Len =< Pos + byte_size(Data) ->
+    binary:part(Data, ReadPos - Pos, Len);
+ra_read(_Pos, _Len, _Ra) ->
+    undefined.
+
+ra_update_size(undefined, _FilterSize, LastDataSize,
+               #ra{on = true, size = Sz} = Ra)
+  when Sz < ?READ_AHEAD_LIMIT andalso
+       LastDataSize < ?READ_AHEAD_LIMIT ->
```

Comment:
Should this really be `LastDataSize < (?READ_AHEAD_LIMIT - FilterSize - ?HEADER_SIZE_B - ?REC_HDR_SZ_SIMPLE_B)`

we don't want to enable read ahead unless we can at least fulfil LastDataSize in the next read ahead op.

for example a single message chunk where the message is 4050 bytes will enable read ahead, but can'tbe fulfilled by 4096 bytes as we need to read the header (48 bytes) and the filter (16 bytes if used) and the length of the record (4 bytes). 4050 + 48 + 16 + 4 > 4096